docs/rules/prefer-destructuring.md

```diff
@@ -0,0 +1,61 @@
+# Prefer destructuring from arrays and objects (prefer-destructuring)
+
+With JavaScript ES6, a new syntax was added for creating variables from an array index of object property, called [destructuring](#further-reading).  This rule enforces usage of destructuring instead of accessing a property through a member expression.
+
+## Rule Details
+
+Examples of **incorrect** code for this rule:
+
+```javascript
+// With `array` enabled
+var foo = array[0];
+
+// With `object` enabled
+var foo = object.foo;
+var foo = object['foo'];
+```
+
+Examples of **correct** code for this rule:
+
+```javascript
+// With `array` enabled
+var [ foo ] = array;
+
+// With `object` enabled
+var { foo } = object;
+```
+
+### Options
+
+This rule takes two properties, `array` and `object`, which can be used to turn on or off the destructuring requirement for each of those types independently.  By default, both are `true`.  If you want to change the behavior, you can configure the rule like so:
```

Comment:
We should also add documentation for the `requireRenaming` option (although it would also be fine to wait and see if we plan to bikeshed on the name).